benchexec/tools/ultimate.py

```diff
@@ -139,39 +139,42 @@ def _query_ultimate_version(self, cmd, api):
             return ""
         stdout = util.decode_to_string(stdout).strip()
         if stderr or process.returncode:
-            logging.warning(
-                "Cannot determine Ultimate version (API %s).\n"
-                "Command was:     %s\n"
-                "Exit code:       %s\n"
-                "Error output:    %s\n"
-                "Standard output: %s",
-                api,
-                " ".join(map(util.escape_string_shell, cmd)),
-                process.returncode,
-                util.decode_to_string(stderr),
-                stdout,
-            )
+            if logging.getLogger().getEffectiveLevel() == logging.DEBUG:
```

Comment:
Hiding only a `logging.debug` statement inside this check doesn't make much sense, right?

Another question is: Does it make sense for the tool-info module to continue if getting the version does not work? Is there a chance that actual benchmarking will work in this case? Otherwise I would simply produce an error here that contains all the relevant information, no need to worry about warnings and debug messages.

If you want to continue, then why not simply have a debug log message with all the raw info first and afterwards a warning if the version cannot be determined?

Reply:
Its just so that either the debug message or the warn message is printed. But I can combine them. 

It makes sense that it continues because it tries different Java versions for different API versions and different wrapper script versions to be downward-compatible. 

The thing is: the Ultimate >=0.2.0 family does only run with Java 11 and higher, the ones before only with Java 8 and lower. But in order to get the version, I have to execute the tool. So I try and find all Java installations and try to execute the tool with all the different APIs. The first that succeeds wins.

Reply:
> Its just so that either the debug message or the warn message is printed. But I can combine them.

I understand, but I would say that having both debug and warn message in case one enables debug messages it not a problem, so the code can be kept simple.